error: 'junction' is not a member of 'std::filesystem::file_type'

e064b5de193c9847f030f320808973c689959562 をMSYS2 MINGW64環境下で、コンパイルしようとすると次のようなエラーが出ます。

```
In file included from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_defs.hpp:21,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_internal_defs.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_port.hpp:24,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_charcode.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_env.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_message.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_result.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_assertion.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_pred.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_internal.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_core_impl.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_result_reporter.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_core.hpp:20,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest.hpp:24,
                 from C:/msys64/home/yumetodo/apache2_log_analyzer/test/test.cpp:4:
C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_printers.hpp: In function 'std::string iutest::detail::FileSystemFileTypeToString(const std::filesystem::file_type&)':
C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/iutest_printers.hpp:454:79: error: 'junction' is not a member of 'std::filesystem::file_type'
  454 |     IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(::std::filesystem::file_type, junction);
      |                                                                               ^~~~~~~~
C:/msys64/home/yumetodo/apache2_log_analyzer/iutest/include/internal/iutest_pp.hpp:84:72: note: in definition of macro 'IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING'
   84 | #define IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(ns, name) case ns::name: return #name
      |                                                                        ^~~~
```

インストールされているgcc関係のパッケージは次のとおりです

```
local/mingw-w64-x86_64-gcc 9.3.0-2 (mingw-w64-x86_64-toolchain)
    GNU Compiler Collection (C,C++,OpenMP) for MinGW-w64
local/mingw-w64-x86_64-gcc-fortran 9.3.0-2 (mingw-w64-x86_64-toolchain)
    GNU Compiler Collection (Fortran) for MinGW-w64
local/mingw-w64-x86_64-gcc-libgfortran 9.3.0-2 (mingw-w64-x86_64-toolchain)
    GNU Compiler Collection (libgfortran) for MinGW-w64
local/mingw-w64-x86_64-gcc-libs 9.3.0-2 (mingw-w64-x86_64-toolchain)
    GNU Compiler Collection (libraries) for MinGW-w64
```

原因は以下の行にあります。これをコメントアウトするとうまくいきました。

https://github.com/srz-zumix/iutest/blob/3afc4249027f5761960bada2d5dcfb9b8525582a/include/iutest_printers.hpp#L453-L455

--- a/include/iutest_printers.hpp
+++ b/include/iutest_printers.hpp
@@ -450,7 +450,7 @@ inline ::std::string FileSystemFileTypeToString(const ::std::filesystem::file_ty
     IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(::std::filesystem::file_type, fifo);
     IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(::std::filesystem::file_type, socket);
     IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(::std::filesystem::file_type, unknown);
-#if defined(IUTEST_OS_WINDOWS)
+#if defined(IUTEST_OS_WINDOWS) && !defined(IUTEST_OS_WINDOWS_MINGW)
     IUTEST_PP_NAMESPACE_ENUM_CASE_RETURN_STRING(::std::filesystem::file_type, junction);
 #endif
     default:

